lib/home_page.dart

```diff
@@ -0,0 +1,314 @@
+import 'package:flutter/material.dart';
+import 'question_model.dart';
+import 'category_filter.dart';
+
+class Home extends StatefulWidget {
+  @override
+  HomeState createState() => HomeState();
+}
+
+class HomeState extends State<Home> {
+  // Questions data. Unfiltered list of questions
+  List data = [];
+  // Will filter the list of questions
+  List questionsListFiltered = [];
+  // These lists will help to filter questions by both search key and category simultaneously
+  // Might be inefficient to use that many lists around, thus, this might change in the future versions to something more dynamic and efficient
+  List questionsListFilteredCategory = [];
+  List questionsListFilteredSearch = [];
+  // isSearching keep track of whether search bar is activated or not. Search bar will appear if user clicked a search icon.
+  bool isSearching = false;
+  // Keeps track of the last chosen category. By default is All Categories.
+  String selectedCategory = "All Categories";
+
+  // Called from widget (class) category filter
+  getCategory(value) {
+    setState(() {
+      filterQuestionsCategory(value);
+    });
+  }
+
+  // Here Eureka will get data from the database.
+  // At this point it contains just dummy lists of questions for test purposes
+  getQuestions() async {
```

Comment:
Please define the type here to be `Future<void>`

Reply:
This caused errors on line 80: _Error: This expression has type 'void' and can't be used._
I tried "Future<dynamic>" instead and I think it worked. Will that be strongly typed?

Reply:
Apologies, I didn't realize that you were returning something with this function. It looks like you're returning `List` so can you try if `Future<List>` will work? If not, then `Future<Dynamic>` is fine.